### PR TITLE
Release Performance Regression 2.0/2.1 [Oct 20] 

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?

**Release Run:  https://buildkite.com/ray-project/release-tests-branch/builds/1135**

Confirmed Regression: 
- benchmarks/many_nodes, throughput [ 8.2 -> 5.2 = 37%]
- benchmarks/many_tasks, NA (couldn't finish) 
- scalability/object_store, latency [ 77 -> 87 = 13% ] 
- scalability/single_node::1000000_queued_time, latency [ 178 -> 212 = 19% ] 
- stress_tests/stress_test_many_tasks, latency [53 -> 74 = 40%] 

Potential Regression 
- benchmarks/many_pgs 
- microbenchmark::single_client_tasks_and_get_batch
- microbenchmark::multi_client_put_gigabytes
- other microbenchmarks? 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/release_logs/2.0.0/stress_tests/stress_test_many_tasks.json [hunk 1]
```diff
@@ -3,45 +3,45 @@
         {
             "perf_metric_name": "stage_0_time",
             "perf_metric_type": "LATENCY",
-            "perf_metric_value": 6.4732019901275635
+            "perf_metric_value": 6.72689962387085
```
Comment (rickyyx) [inline release/release_logs/2.0.0/stress_tests/stress_test_many_tasks.json:6]: definitely seeing regression - will run again when `many_tasks` failure resolved.

File: release/release_logs/2.0.0/benchmarks/many_nodes.json [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
             "perf_metric_name": "tasks_per_second",
             "perf_metric_type": "THROUGHPUT",
-            "perf_metric_value": 8.153990032133354
+            "perf_metric_value": 5.207718176892282
         }
     ],
     "success": "1",
-    "tasks_per_second": 8.153990032133354,
-    "time": 422.63934540748596
+    "tasks_per_second": 5.207718176892282,
```
Comment (rickyyx) [inline release/release_logs/2.0.0/benchmarks/many_nodes.json:13]: Slight regression on tasks - probably related to the `many_tasks` failure.

File: release/release_logs/2.0.0/benchmarks/many_pgs.json [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
     "num_pgs": 1000,
     "perf_metrics": [
         {
             "perf_metric_name": "pgs_per_second",
             "perf_metric_type": "THROUGHPUT",
-            "perf_metric_value": 16.867906637394082
+            "perf_metric_value": 15.991731326176165
```
Comment (rickyyx) [inline release/release_logs/2.0.0/benchmarks/many_pgs.json:9]: Looks like variance too me.

File: release/release_logs/2.0.0/microbenchmark.json [hunk 4]
```diff
@@ -1,274 +1,274 @@
 {
     "1_1_actor_calls_async": [
-        6146.424440513006,
-        128.76132098689908
+        5527.366120616061,
```
Comment (rickyyx) [inline release/release_logs/2.0.0/microbenchmark.json:3]: What are these values actually?

Comment (rickyyx) [reply release/release_logs/2.0.0/microbenchmark.json:3]: e.g. throughput or time?

File: release/release_logs/2.0.0/microbenchmark.json [hunk 5]
```diff
@@ -1,274 +1,274 @@
...
             "perf_metric_name": "single_client_put_gigabytes",
             "perf_metric_type": "THROUGHPUT",
-            "perf_metric_value": 19.66716992775968
+            "perf_metric_value": 19.34338494332012
         },
         {
             "perf_metric_name": "single_client_tasks_and_get_batch",
             "perf_metric_type": "THROUGHPUT",
-            "perf_metric_value": 12.659666662711272
+            "perf_metric_value": 10.102375809036388
```
Comment (rickyyx) [inline release/release_logs/2.0.0/microbenchmark.json:114]: looks like task regression.

File: release/release_logs/2.0.0/microbenchmark.json [hunk 6]
```diff
@@ -1,274 +1,274 @@
...
         },
         {
             "perf_metric_name": "multi_client_put_gigabytes",
             "perf_metric_type": "THROUGHPUT",
-            "perf_metric_value": 34.625799191250174
+            "perf_metric_value": 28.928286311767714
```
Comment (rickyyx) [inline release/release_logs/2.0.0/microbenchmark.json:119]: This looks like a regression.

File: release/release_logs/2.0.0/microbenchmark.json [hunk 7]
```diff
@@ -1,274 +1,274 @@
...
             "perf_metric_name": "single_client_tasks_sync",
             "perf_metric_type": "THROUGHPUT",
-            "perf_metric_value": 1424.3365828460285
+            "perf_metric_value": 1183.6629906022138
         },
         {
             "perf_metric_name": "single_client_tasks_async",
             "perf_metric_type": "THROUGHPUT",
-            "perf_metric_value": 13149.808089478229
+            "perf_metric_value": 10247.298811196548
```
Comment (rickyyx) [inline release/release_logs/2.0.0/microbenchmark.json:134]: all related to task regresiion likely

File: release/release_logs/2.0.0/scalability/single_node.json [hunk 8]
```diff
@@ -11,30 +11,30 @@
...
             "perf_metric_name": "10000_get_time",
             "perf_metric_type": "LATENCY",
-            "perf_metric_value": 26.833145973
+            "perf_metric_value": 26.702664286000015
         },
         {
             "perf_metric_name": "1000000_queued_time",
             "perf_metric_type": "LATENCY",
-            "perf_metric_value": 178.32711092900007
+            "perf_metric_value": 212.33255058699996
```
Comment (rickyyx) [inline release/release_logs/2.0.0/scalability/single_node.json:29]: This seems to be a regression.

File: release/release_logs/2.0.0/scalability/object_store.json [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
-    "broadcast_time": 76.73496933199999,
+    "broadcast_time": 86.96337815599998,
     "num_nodes": 50,
     "object_size": 1073741824,
     "perf_metrics": [
         {
             "perf_metric_name": "time_to_broadcast_1073741824_bytes_to_50_nodes",
             "perf_metric_type": "LATENCY",
-            "perf_metric_value": 76.73496933199999
+            "perf_metric_value": 86.96337815599998
```
Comment (rickyyx) [inline release/release_logs/2.0.0/scalability/object_store.json:9]: looks like a regression

File: release/release_logs/2.0.0/stress_tests/stress_test_many_tasks.json [hunk 10]
```diff
@@ -3,45 +3,45 @@
...
             "perf_metric_name": "stage_1_avg_iteration_time",
             "perf_metric_type": "LATENCY",
-            "perf_metric_value": 20.34087462425232
+            "perf_metric_value": 24.53009397983551
         },
         {
             "perf_metric_name": "stage_2_avg_iteration_time",
             "perf_metric_type": "LATENCY",
-            "perf_metric_value": 53.219086503982545
+            "perf_metric_value": 73.92530932426453
```
Comment (rickyyx) [inline release/release_logs/2.0.0/stress_tests/stress_test_many_tasks.json:16]: regression.

File: release/release_logs/2.0.0/microbenchmark.json (hunk 4 above)
Comment (rickyyx) [reply release/release_logs/2.0.0/microbenchmark.json:3]: cc @rkooo567

Comment (jjyao) [reply release/release_logs/2.0.0/microbenchmark.json:3]: It's throughput. The second number is standard deviation.

File: release/release_logs/2.0.0/benchmarks/many_pgs.json (hunk 3 above)
Comment (jjyao) [reply release/release_logs/2.0.0/benchmarks/many_pgs.json:9]: yes.

**Comment by rkooo567** [comment]
Looks like most of significant regression has something to do with many_tasks failure (it is a really good test lol)

**Comment by rkooo567** [comment]
We should rerun tests after https://github.com/ray-project/ray/pull/29517

**Pull request closed without merging**